### PR TITLE
Update documentation to reflect automated e2e test artifacts.

### PR DESCRIPTION
**Pull request by AlgoStephenAkiki**
Removed documentation for manual builds since this is now automated.

## Summary

Explain the goal of this change and what problem it is solving. Format this cleanly so that it may be used for a commit message, as your changes will be squash-merged.

## Test Plan

How did you test these changes? Please provide the exact scenarios you tested in as much detail as possible including commands, output and rationale.

### CODE REVIEW

File: misc/README.md [hunk 1]
```diff
@@ -2,15 +2,6 @@
 
 `make e2e` runs tests with input generated by [go-algorand `e2e_subs` tests](https://github.com/algorand/go-algorand/blob/master/test/scripts/e2e_client_runner.py).  As of writing `make e2e` is the _only_ automated validation of indexer's consumption of algod output.
 
-The process for updating the `make e2e`'s input is manual.  Each time a go-algorand E2E test is modified, the test modifier _must_ generate new artifacts for `make e2e` consumption.  Here's the process:
```
[the remark this answers is not in the document]
Comment (winder) [reply misc/README.md:5]: I'm expecting the nightly test to notify on a regression / incompatibility automatically. Previously we relied on people remembering to update the artifacts.

AFAIK most people don't know about the indexer E2E testing, so this documentation probably wasn't being followed.

[the remark this answers is not in the document]
Comment (michaeldiamant) [reply misc/README.md:5]: @winder Got it - When the automated test fails, how do you expect folks to reproduce it locally?

The (deleted) instructions provide an expedited way to test locally.  In my experience, when trying to iterate on a broken test, `make e2e` is prohibitively expensive.

That said - I consider myself the least knowledgeable person in the discussion.  If you're good with the PR as is, then I'm also good with it.

File: misc/README.md [hunk 2]
```diff
@@ -2,15 +2,6 @@
...
-
 ## e2elive.py prerequisites
 `e2elive.py` requires access to algorand-indexer and goal binaries. If you receive an error related to the algorand-indexer binary then you need to build the indexer from scratch by running `make` in the root of the indexer repository. This will insert a binary in the `cmd/algorand-indexer` directory, where `e2elive.py` expects it by default.
```
Comment (michaeldiamant) [inline misc/README.md:6]: @AlgoStephenAkiki _If_ https://github.com/algorand/indexer/pull/956#discussion_r847871513 is resolved without further changes to the PR, then I think the highlighted section can also be removed.  It's only referenced by the deleted instructions.

```suggestion

```

File: misc/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (winder) [reply misc/README.md:5]: @michaeldiamant what do you think about a policy where you should monitor the algod AND indexer nightly builds (or #dev-alerts) after having changes merged.

[the remark this answers is not in the document]
Comment (michaeldiamant) [reply misc/README.md:5]: @winder Thoughts inline below.

> a policy where you should monitor the algod AND indexer nightly builds (or #dev-alerts) after having changes merged.

* I have mixed feelings.  I think it feels like a good idea, but in practice is difficult to enforce.  And I personally dislike passively monitoring channels.
* Idea:  In my experience, finding a way to assign ownership (ideally, 1 person) produced better outcomes.  For example, it's possible to imagine tooling that @ mentions anyone with a commit after a failed nightly build.  
  * The idea makes it harder to ignore a failure without requiring passive channel monitoring. 
  * The idea is obviously out-of-scope to the PR.  Only brainstorming.
* For the PR's purpose, I'm on-board to include expectations verbiage.  While I have some reservations, I think the suggestion improves the status quo.

[the remark this answers is not in the document]
Comment (winder) [reply misc/README.md:5]: @AlgoStephenAkiki we should probably keep the instructions for running `e2elive` locally.

[the remark this answers is not in the document]
Comment (AlgoStephenAkiki) [reply misc/README.md:5]: Sure.

**Review by michaeldiamant — APPROVED** [review]
@AlgoStephenAkiki Thanks for taking the time to update indexer E2E instructions as part of https://github.com/algorand/go-algorand/pull/3849.  ☕

**Review by winder — APPROVED** [review]

**Pull request merged**